src/App/PropertyLinks.cpp

```diff
@@ -2519,8 +2520,78 @@ class App::DocInfo :
     App::Document *pcDoc{nullptr};
     std::set<PropertyXLink*> links;
 
-    static std::string getDocPath(
-            const char *filename, App::Document *pDoc, bool relative, QString *fullPath = nullptr)
+    struct PathInfo {
+        QDir dir;
+        QString postfix;
+    };
+
+    static QString resolvePath(const QString &path, QFileInfo docPath)
+    {
+        std::set<QString> visited;
+        std::vector<PathInfo> pendings;
+        QDir docDir(docPath.dir());
+        pendings.push_back({docDir, QString()});
+        visited.insert(docPath.path());
+
+        // Handle the case where the owner document is a symlink
+        while (true) {
+            QString target = docPath.symLinkTarget();
+            if (target.isEmpty() || !visited.insert(target).second)
+                break;
+            docPath = QFileInfo(target);
+            docDir = docPath.dir();
+            pendings.push_back({docDir, QString()});
+        }
+
+        for (size_t i=0; i<pendings.size(); ) {
+            // Make copy instead of reference to the entry as we may push back
+            // new entries later which invalidates the reference.
+            PathInfo entry = pendings[i];
+            QDir dir(entry.dir.absoluteFilePath(entry.postfix));
+            QString testPath = QDir::cleanPath(dir.filePath(path));
+            if (QFileInfo(testPath).exists())

```

Comment:
`Use the static QFileInfo::exists() instead. It's documented to be faster. [-Wclazy-qfileinfo-exists]`